src/neuroconv/utils/dict.py

```diff
@@ -198,3 +198,24 @@ def dict_deep_update(
             dict_to_update[key_to_update] = update_values
 
     return dict_to_update
+
+
+class DeepDict(dict):
+    """A defaultdict of defaultdicts"""
```

Comment:
Would it make sense, perhaps not in this PR, but in a follow-up; to define a `.deep_update` method to this class which is essentially our current `deep_dict_update` or the one from FastAPI?

Reply:
yes, I think it would. I can put that in here. Also, I could work on a better __init__